### ISSUE TITLE
GNU parallel causes "Unsupported use of '&&'. In fish, please use 'COMMAND; and COMMAND'."

### ISSUE THREAD

**Issue by frankseide** [fish-shell/fish-shell#5582]
Hi, I am not able to run GNU parallel from fish. It throws a number of random errors that seem to have nothing to do with the command I am running.

Not clear if this is a dup of Issue #1084, but the latest fish via `apt-get install` on Ubuntu 16 has this problem.

bash:
```
$ parallel --pipe -k -j16 "tr '[a-z]' '[A-Z]'"
a                                   # I enter these two lines by hand, then Ctrl-D
b 
<Ctrl-D>
A                                   # this is the output from parallel
B
```
fish:
```
> parallel --pipe -k -j16 "tr '[a-z]' '[A-Z]'"
a
b 
<Ctrl-D>
Unsupported use of '&&'. In fish, please use 'COMMAND; and COMMAND'.
fish:              test ! -s "/tmp/pargWeVs.chr" && rm -f "/tmp/pargWeVs.chr" && exec true;
                                                  ^
Unsupported use of '&&'. In fish, please use 'COMMAND; and COMMAND'.
fish:              test ! -s "/tmp/parbL67U.chr" && rm -f "/tmp/parbL67U.chr" && exec true;
...
```
I also tried to run this from fish via bash (bash -c ...) and by setting SHELL to `/bin/bash` or `bash`, same problem.

fish, version 2.2.0
2.2.0
Linux MT-GPU-009.redmond.corp.microsoft.com 4.4.0-116-generic #140-Ubuntu SMP Mon Feb 12 21:23:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
xterm

same after `sh -c 'env HOME=$(mktemp -d) fish'`

**Comment by faho** [comment]
>fish, version 2.2.0

Your fish version is quite old and unsupported.

Upgrade to fish 3.0, which added support for that syntax, e.g. via [our PPA](https://launchpad.net/~fish-shell/+archive/ubuntu/release-3), which should be available for Ubuntu down to 16.04 (xenial), which IIRC is what WSL is currently based on.

However, in general it's not recommended to run arbitrary shell scripts with fish as it's not generally POSIX compatible. So parallel is doing something it shouldn't do here, which is use $SHELL (or the entry in /etc/passwd) to execute "the user's shell". Instead, it should write the script to be compatible with e.g. /bin/sh, and then use /bin/sh, or bash and use bash, or ....

**Comment by faho** [comment]
I just tested your script and it works in 3.0.0, so upgrading should fix this particular use.

**Comment by frankseide** [comment]
Thanks for the quick reply! I installed `fish, version 3.0.0`, but I now get a different error:
```
> parallel --pipe -k -j16 "tr '[a-z]' '[A-Z]'"
a
b
<Ctrl-D>
fish: Command substitutions not allowed
```

**Comment by frankseide** [comment]
Unfortunately it does not solve it:
```
> function parallel1
                           set -lx SHELL bash
                           command parallel $argv
                     end
> parallel1 --pipe -k -j16 "tr '[a-z]' '[A-Z]'"
a
b
<Ctrl-D>
fish: Command substitutions not allowed
```
Nor does this, which I believe is equivalent:
```
env SHELL=bash parallel --pipe -k -j16 "tr '[a-z]' '[A-Z]'"
a
b
fish: Command substitutions not allowed
```
I noticed that `parallel` is actually a perl script, which has shell interaction and a reference to a "parent shell." I don't have time to further investigate this, my hunch is that it tries to execute a bash command but somehow gets fish instead. So the bug is not in fish, but in `parallel` not not explicitly calling bash.

I found an alternative solution by wrapping `parallel` in a bash script that I invoke from fish. So I am no longer blocked, and `parallel`'s idiosyncrasies are best encapsulated away anyway. Thanks for your help! Feel free to close this Issue. Thanks!
